macformula/ecu/lvcontroller/client.go

```diff
@@ -0,0 +1,136 @@
+package lvcontroller
+
+import (
+	"go.uber.org/zap"
+
+	"github.com/macformula/hil/macformula/pinout"
+	"github.com/pkg/errors"
+)
+
+const (
+	_clientName = "lv_controller_client"
+)
+
+type Client struct {
+	l             *zap.Logger
+	pinController *pinout.Controller
+}
+
+func NewClient(pc *pinout.Controller, l *zap.Logger) *Client {
+	return &Client{
+		l:             l.Named(_clientName),
+		pinController: pc,
+	}
+}
+
+func (c *Client) ReadTsalEn() (bool, error) {
+	lvl, err := c.pinController.ReadDigitalLevel(pinout.TsalEn)
+	if err != nil {
+		return false, errors.Wrap(err, "read digital level")
+	}
+
+	return lvl, nil
+}
+
+func (c *Client) ReadRaspiOn() (bool, error) {
+	lvl, err := c.pinController.ReadDigitalLevel(pinout.RaspiEn)
+	if err != nil {
+		return false, errors.Wrap(err, "read digital level")
+	}
+
+	return lvl, nil
+}
+func (c *Client) ReadFrontControllerOn() (bool, error) {
+	lvl, err := c.pinController.ReadDigitalLevel(pinout.FrontControllerEn)
+	if err != nil {
+		return false, errors.Wrap(err, "read digital level")
+	}
+
+	return lvl, nil
+}
+
+func (c *Client) ReadSpeedgoatOn() (bool, error) {
+	lvl, err := c.pinController.ReadDigitalLevel(pinout.SpeedgoatEn)
+	if err != nil {
+		return false, errors.Wrap(err, "read digital level")
+	}
+
+	return lvl, nil
+}
+
+func (c *Client) ReadAccumulatorOn() (bool, error) {
+	lvl, err := c.pinController.ReadDigitalLevel(pinout.AccumulatorEn)
+	if err != nil {
+		return false, errors.Wrap(err, "read digital level")
+	}
+
+	return lvl, nil
+}
+
+func (c *Client) ReadMotorControllerPrechargeOn() (bool, error) {
+	lvl, err := c.pinController.ReadDigitalLevel(pinout.MotorControllerPrechargeEn)
+	if err != nil {
+		return false, errors.Wrap(err, "read digital level")
+	}
+
+	return lvl, nil
+}
+
+func (c *Client) ReadMotorControllerPrechargeOff() (bool, error) {
+	lvl, err := c.ReadMotorControllerPrechargeOn()
+	return !lvl, err
+}
```

Comment:
We should just keep this to the ReadMotorControllerPrechargeOn() function

Reply:
The reason I had this is for the poll utility in the lvstartup test which takes a `func() (bool, error)` and returns when it reads a true. I'm not sure how useful these ecu clients really are apart from the CAN stuff. We should re-evaluate if we really need them.

Reply:
I fixed this by adding functional options to the poll function